apps/web/src/stores/documentVersions.ts

```diff
@@ -111,8 +119,46 @@ export default function useDocumentVersions(
         }
       }
     },
-    [executeDestroyDocument, mutate, data],
+    [executeDestroyDocument, mutate, data, currentDocument?.documentUuid],
+  )
+
+  const destroyFolder = useCallback(
+    async (path: string) => {
+      const [_, error] = await executeDestroyFolder({
+        projectId: project.id,
+        commitId: commit.id,
+        path,
+      })
+
+      if (error) {
+        toast({
+          title: 'Error deleting folder',
+          description: error.formErrors?.[0] || error.message,
+          variant: 'destructive',
+        })
+      } else {
+        mutate()
+
+        if (currentDocument?.path?.startsWith?.(path)) {
+          router.push(
+            ROUTES.projects
+              .detail({ id: project.id })
+              .commits.detail({ uuid: commit.uuid }).documents.root,
+          )
+        }
```

Comment:
Edge case: Add a final `/` to `path` if it doesn't have it already; otherwise you'll be redirecting the user when removing a folder with the same name and path as another file.

```
a/
  b    <-- user is here
  b/   <-- user removes this folder
```

Reply:
This is `/documents/` path is not a folder path